### PR TITLE
Build Braid as an integration test

### PR DESCRIPTION
**Pull request by mrm007**
As part of this, I migrated `scripts/braid-fixture.sh` to TypeScript and made `scripts` a package in the monorepo. This allows us to have an isolated TypeScript environment with modules, top-level `await`, `--moduleResolution bundler` and more.

The monorepo was also upgraded to TypeScript 5 which looks like a seamless upgrade since we're not affected by any of the breaking changes.

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -142,7 +142,8 @@
     "memfs": "^3.4.11",
     "strip-ansi": "^7.0.1",
     "sync-dependencies": "^1.0.4",
-    "typescript": "*"
+    "typescript": "*",
+    "unbuild": "^1.1.2"
```
Comment (mrm007) [inline packages/core/package.json:146]: This is not a new dependency. I moved it here from the root of the monorepo.

File: fixtures/with-side-effects/src/lib/atoms/atoms.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import * as resetStyles from '../reset/reset.css';
 import type { RequiredResponsiveValue } from './sprinkles.css';
 import { sprinkles } from './sprinkles.css';
+import * as resetStyles from '../reset/reset.css';
```
Comment (askoufis) [inline fixtures/with-side-effects/src/lib/atoms/atoms.ts:3]: Should this stay above `sprinkles`, or does it not matter?

File: scripts/braid-fixture.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    `git clone --depth=1 --no-checkout -b ${branch} ${repo} ${submodule}`,
+  );
+}
+
+if (argv.absorb) {
+  //  Add as a submodule
+  await run(`git submodule add -b ${branch} --force ${repo} ${submodule}`);
+
+  //  Move the .git dir from $fixture/.git into parent repo's .git
+  await run(`git submodule absorbgitdirs`);
```
Comment (askoufis) [inline scripts/braid-fixture.ts:65]: This potentially quite destructive right? I know that `--absorb` is only set on CI, but I'd feel more confident if this was behind a `process.env.CI` check as well.

File: scripts/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,105 @@
+{
+  "extends": "../tsconfig.json",
+  "exclude": ["node_modules"], // cancel out the exclude from the root tsconfig
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig to read more about this file */
```
Comment (askoufis) [inline scripts/tsconfig.json:5]: Can we remove all the commented lines from this file?

File: fixtures/with-side-effects/src/lib/atoms/atoms.ts (hunk 2 above)
Comment (mrm007) [reply fixtures/with-side-effects/src/lib/atoms/atoms.ts:3]: This is caused by [eslint-plugin-import >= 2.27.5](https://github.com/import-js/eslint-plugin-import/blob/main/CHANGELOG.md#2275---2023-01-16)

Same issue reportei first noticed here https://github.com/seek-oss/sku/pull/735#discussion_r1102178164

File: scripts/tsconfig.json (hunk 4 above)
Comment (mrm007) [reply scripts/tsconfig.json:5]: It doesn't really bother me, and it's nice to have access to all the options without having to look them up.

File: scripts/braid-fixture.ts (hunk 3 above)
Comment (mrm007) [reply scripts/braid-fixture.ts:65]: It's not destructive, but it's behind a flag in case you don't want to do that again if you've done it before.

Comment (askoufis) [reply scripts/braid-fixture.ts:65]: Oh, I read the comment and assumed it was moving things around within `.git`, but it's just a submodule command 🤦

**Review by askoufis — APPROVED** [review]

**Pull request merged**